tests/rptest/tests/broken_connection_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import time
+import random
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import RESTART_LOG_ALLOW_LIST
+from rptest.clients.types import TopicSpec
+from rptest.tests.end_to_end import EndToEndTest
+from ducktape.mark import matrix
+
+
+class BrokenConnectionTest(EndToEndTest):
+    def __init__(self, ctx, *args, **kwargs):
+        super(BrokenConnectionTest, self).__init__(
+            ctx,
+            *args,
+            extra_rp_conf={
+                'enable_leader_balancer': False
+            },
+            **kwargs)
+        self._ctx = ctx
+
+    def _node_ip(self, node):
+        out = self.redpanda.nodes[0].account.ssh_output(f"getent hosts {node.name}")
+        return out.split()[0].decode()
+
+    def _node_ips(self):
+        return [(node, self._node_ip(node)) for node in self.redpanda.nodes]
+
+    def _make_rule(self, ip):
+        reject_types = ["icmp-host-unreachable", "tcp-reset"]
```

Comment:
Can use deterministic faults instead of relying on randomness? It's hard to debug if something go wrong.

Reply:
yes, though, there is always a record in the log of the actions that lead up to something going wrong.

i'll add separate tests that do individual actions in isolation. great idea.

but having the randomness lets us explore a wider variety of scenarios.

Reply:
Can it be seeded + log the seed?  I know this is a pain in python's stdlib, but maybe if necessary we just copy paste a basic PRNG that we can control so that we can repro tests with a known seed.  I'm also guilty of having not already done this.

Reply:
yeh i think so. i'll see what i can up with